generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -1227,6 +1238,20 @@ private String fieldDefault(FieldSpec field) {
                 }
                 return field.defaultString() + "L";
             }
+        } else if (field.type() instanceof FieldType.UUIDFieldType) {
+            if (field.defaultString().isEmpty()) {
```

Comment:
UUIDs can never be null.  Therefore, it doesn't make sense to use null as a default.  Even if it did, we don't choose null as a default for anything unless the user selects it specifically by writing `"default": "null"`.